developers_chamber/ecs_utils.py

```diff
@@ -437,18 +478,58 @@ def run_task_and_wait_for_success(cluster, task_definition, command, name, succe
 
 def get_services_arns(cluster, region):
     ecs_client = _get_ecs_client(region)
-    resp = ecs_client.list_services(cluster=cluster)
+
+    try:
+        resp = ecs_client.list_services(cluster=cluster)
+    except ecs_client.exceptions.ClusterNotFoundException:
+        raise ClickException('Cluster not found: \'{}\''.format(cluster))
+    except ClientError as ex:
+        raise ClickException(ex)
+
     services_arns = resp['serviceArns']
     return services_arns
 
 
 def get_tasks_for_service(cluster, service, region):
     ecs_client = _get_ecs_client(region)
-    resp = ecs_client.list_tasks(cluster=cluster, serviceName=service)
+
+    try:
+        resp = ecs_client.list_tasks(cluster=cluster, serviceName=service)
+    except ecs_client.exceptions.ClusterNotFoundException:
+        raise ClickException('Cluster not found: \'{}\''.format(cluster))
+    except ecs_client.exceptions.ServiceNotFoundException:
+        raise ClickException('Service not found: \'{}\''.format(service))
+    except ClientError as ex:
+        raise ClickException(ex)
+
     tasks_arns = resp['taskArns']
     return tasks_arns
 
 
+def t_stop_services_and_wait_for_tasks_to_stop(cluster, services, timeout, region):
+    ecs_client = _get_ecs_client(region)
+
+    tasks = []
+
+    for service in services:
+        service_tasks = get_tasks_for_service(cluster=cluster, service=service, region=region)
+        if not service_tasks:
+            LOGGER.info('No active tasks found in service \'{}\''.format(service))
+            continue
+        tasks += service_tasks
+        stop_service(cluster=cluster, service=service, region=region)
+
+    threads = []
+
+    for task in tasks:
```

Comment:
will be better to use thread pool https://www.codementor.io/lance/simple-parallelism-in-python-du107klle

Reply:
what is the problem with threading? As the AWS API calls are I/O blocking, it's better to run more threads than the number of available cores. BTW this PR is WIP - the actual PR adding threads is here: https://github.com/druids/developers-chamber/pull/32

Reply:
Have you considered asyncio?

Reply:
yes and it's overkill for this easy task

Reply:
Also I am not sure that async ECS Client is supported by boto3

Reply:
fixed in #32